source/extensions/filters/udp/udp_proxy/hash_policy_impl.cc

```diff
@@ -0,0 +1,38 @@
+#include "extensions/filters/udp/udp_proxy/hash_policy_impl.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace UdpProxy {
+
+class SourceIpHashMethod : public HashPolicyImpl::HashMethod {
+public:
+  uint64_t evaluate(const Network::Address::Instance& downstream_addr) const override {
+    ASSERT(downstream_addr.ip() != nullptr);
+    ASSERT(!downstream_addr.ip()->addressAsString().empty());
+    return HashUtil::xxHash64(downstream_addr.ip()->addressAsString());
```

Comment:
Now, there are no error cases because all abnormal cases are checked by assert.
So, simply returns hash value not optional.